xgcm/grid.py

```diff
@@ -941,6 +941,27 @@ def _assign_metrics(self, metrics):
                 # check for duplicate dimensions among each axis metric
                 self._metrics[metric_axes].append(metric_var)
 
+    def _check_axes(self, axis):
+        """Checks if axis(or axes) in `axis` are part of the grid"""
+        check_axes = [ax for ax in axis if ax not in self.axes]
+        check_axes_str = ",".join(check_axes)
+        if any(check_axes):
+            raise ValueError("Axis %s not found in grid object" % check_axes_str)
+
+    def _get_dims_from_axis(self, da, axis):
+        dim = []
+        for ax in axis:
+            all_dim = self.axes[ax].coords.values()
+            matching_dim = [di for di in all_dim if di in da.dims]
+            if len(matching_dim) == 1:
+                dim.append(matching_dim[0])
+            else:
+                raise ValueError(
+                    "Did not find single matching dimension corresponding to axis %s. Got (%s)"
+                    % (ax, matching_dim)
+                )
+        return dim
```

Comment:
This makes sense to split out because we call it twice and it's pretty complicated.

Reply:
And I will reuse it even more often for the other ones.